src/prefect/utilities/context.py

```diff
@@ -93,11 +93,15 @@ def __call__(self, *args: MutableMapping, **kwargs: Any) -> Iterator["Context"]:
         """
         previous_context = self.copy()
         try:
-            self.update(*args, **kwargs)
+            new_context = Context(*args, **kwargs)
+            if "config" in new_context:
+                new_context["config"] = as_nested_dict(new_context["config"])
+            self.update(new_context)  # type: ignore
```

Comment:
This will completely overwrite the existing config with whatever is passed to this context. Perhaps we could only update new keys:

```python
new_context['config'] = as_nested_dict(collections.merge_dicts(self.get('config', {}), new_context.get('config', {})), Config)
```

Reply:
As a Prefect user, I could imagine a scenario where we might want different settings in different environments, e.g. for external Dask Cluster #1 disable shipping logs to cloud, for external Dask Cluster #2 shipping logs to cloud is fine so I would vote for only updating new keys. At a minimum, if this does overwrite config it would be nice to have visibility into that behavior via a WARN log message (something like "WARN overwriting config with values from context -- local config settings will not be used") and perhaps a DEBUG message with details ("DEBUG new config settings from context:\nkey1 = val1, etc.") BTW, overall I love this change so a big +1 for me.

Reply:
@joeschmid with the benefit of actually drinking my coffee (and talking with @cicdw) I realized that the entire `config` object is going to be shipped from "machine 1" to "machine 2" so fine-grained opportunities to control overwritten keys would be lost.

We're exploring a more targeted approach in which only the "cloud" section of the config would be considered mutable and shipped to remote workers. If shipping configs becomes truly useful, then we could seek to expand the shipped keys or allow users a way to define what keys they wanted shipped. Stay tuned more updates soon!

Reply:
Yup, without giving away all the details, this change (which should be released in 0.6.4) coupled with another that will be released in 0.6.5 will allow you to easily specify which Flows should / should not log to cloud by "labeling" the Flow object itself appropriately!